### PR TITLE
Plan student on training + slots (DD-580)

### PR DESCRIPTION
**Pull request by hngl**

### CODE REVIEW

File: includes/RestApiClient/Model/Address.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dation\Woocommerce\RestApiClient\Model;
```
Comment (Liiva) [inline includes/RestApiClient/Model/Address.php]: Models in the RestApiClient namespace feels a bit weird, I realise they are representations of the models we provide in our API but they're still just DTO classes usable in the entire codebase.

**Comment by Liiva** [comment]
This feels nasty, I would replace this with `TranslatorInterface` which has a `translate` function, then wrap the `__` call in a class like `WordpressTranslator` implementing that interface. This isolates the "nastyness" in a single class and allows for easy dependency injection later on (if needed)

**Comment by Liiva** [comment]
Looking at the `translate` calls, you pass already translated strings. How does the translate mechanism actually work? Normally you would pass it a certain key, for example `SYNC_FAILED` and then it would read the translation from somewhere depending on your locale (in this case 'Het synchroniseren met Dation is mislukt')

**Comment by Liiva** [comment]
This fetches metadata about a POST request? Please see whether you can apply an interface here (similar to the translator-comment) as it's not clear from the code what this is supposed to do.

File: includes/NormalizerFactory.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Dation\Woocommerce;
+
+use Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+use Symfony\Component\PropertyInfo\PropertyInfoExtractor;
```
Comment (Liiva) [inline includes/NormalizerFactory.php]: As you never seem to actually set `self::$normalizer` this statement is always true

File: includes/NormalizerFactory.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?php
```
Comment (Liiva) [inline includes/NormalizerFactory.php]: |`line 10`|
|---|
I think this should be either named ObjectNormalizerFactory, or it should return a NormalizerInterface

File: includes/RestApiClient/RestApiClient.php [hunk 4]
```diff
@@ -5,8 +5,15 @@
 namespace Dation\Woocommerce\RestApiClient;
 
 use DateTime;
```
Comment (Liiva) [inline includes/RestApiClient/RestApiClient.php]: The lack of dependency injection is quite apparent in these 2 lines. How hard would it be to add the DI-component from Symfony?

**Comment by Liiva** [comment]
|`includes/RestApiClient/RestApiClient.php:99`|
|---|
Can we just pass the request content to the serializer? What if the POST fails because of a validation error? (the same kind of applies to the other requests made here)

**Comment by Dirkhuethorst** [comment]
Testing experience shows that guzzle throws an error before that can happen. The error is caught and handled in OrderManager::sendToDation or OrderManager::synchronizeEnrollment

**Comment by Dirkhuethorst** [comment]
(it fetches meta data about a wp 'post' object, which can be pretty much anything.)

File: includes/RestApiClient/Model/Address.php (hunk 1 above)
Comment (Dirkhuethorst) [reply includes/RestApiClient/Model/Address.php]: Moved from includes/RestApiClient/Model to includes/Model

**Comment by Liiva** [comment]
Ok, if we know what kind of exceptions are thrown then it might make sense to add `@throws SomeException` phpdoc. Other than that this can be resolved

File: includes/RestApiClient/RestApiClient.php (hunk 4 above)
Comment (Dirkhuethorst) [reply includes/RestApiClient/RestApiClient.php]: Maybe next iteration?

Comment (Dirkhuethorst) [reply includes/RestApiClient/RestApiClient.php]: Maybe next iteration?

**Pull request merged**